arch/sim/src/cmake/Toolchain.cmake

```diff
@@ -94,6 +94,11 @@ else()
   endif()
 endif()
 
+if(CONFIG_DEBUG_OPT_UNUSED_SECTIONS)
+  add_link_options(-Wl,--gc-sections)
+  add_compile_options(-ffunction-sections -fdata-sections)
```

Comment:
what's about Makefile

Reply:
Makefile has these flags but CMake not
https://github.com/apache/nuttx/blob/master/boards/sim/sim/sim/scripts/Make.defs#L117-L118
https://github.com/apache/nuttx/blob/master/boards/sim/sim/sim/scripts/Make.defs#L279